numpy/lib/function_base.py

```diff
@@ -2669,6 +2689,29 @@ class vectorize(object):
            <http://docs.scipy.org/doc/numpy/reference/c-api.generalized-ufuncs.html>`_.
     """
 
+    # __new__ is overriden as a clean way to provide decorator-with-keywords
+    # syntax while allowing 'vectorize' to remain as the type of vectorized
+    # functions.
+    def __new__(cls, pyfunc=None, *args, **kwargs):
+
+        # For backwards compatibility, subclasses must not use the overloaded
+        # __new__, as vectorize.__new__ and subclass.__init__ will receive the
+        # same arguments.
+        if cls is not vectorize:
```

Comment:
Hmm, I see now that my suggestion might well break at least one test (renaming of `pyfunc`). Would it be an idea to be stricter with when the decorator is returned, in that there should be no positional arguements at all? In that case, one might do:
```
if args or 'pyfunc' in kwargs:
    return object.__new__(cls)
```
(I think doing introspection on `__init__` for the name of the first argument probably goes a bit too far...)